app/jobs/solr_reindex_all_job.rb

```diff
@@ -3,13 +3,26 @@
 class SolrReindexAllJob < ApplicationJob
   queue_as :solr_index
 
-  def perform
+  def self.job_limit
+    5000
+  end
+
+  def self.solr_batch_limit
+    500
+  end
+
+  def perform(start_position = 0)
     solr = SolrService.connection
-    # Groups of 500
-    ParentObject.find_in_batches do |group|
-      solr.add(group.map(&:to_solr).compact)
-      solr.commit
+    # Groups of limit
+    parent_objects = ParentObject.order(:oid).offset(start_position).limit(SolrReindexAllJob.job_limit)
+    last_job = parent_objects.count < SolrReindexAllJob.job_limit
```

Comment:
@martinlovell is there a point that `last_job` would ever return false since we're already limiting `parent_objects` to 5000?

Reply:
nevermind! figured it out. 🙂